integration/benchmark/cmd/ci/main.go

```diff
@@ -163,57 +170,56 @@ func main() {
 	if err != nil {
 		log.Fatal().Err(err).Msgf("unable to init loader")
 	}
+
+	// run load
 	lg.Start()
 
-	testStartTime := time.Now()
+	// prepare data slices
+	dataSlices := make([]dataSlice, 0)
+	go func() {
+		leadDuration, _ := time.ParseDuration(*leadTime)
+		sliceDuration, _ := time.ParseDuration(*sliceSize)
+
+		// wait through lead duration.
+		time.Sleep(leadDuration)
+
+		// grab time into start time
+		startTime := time.Now()
+		// grab total executed transactions into start transactions
+		startExecutedTransactions := lg.GetTxExecuted()
+
+		for !lg.Stopped {
```

Comment:
This is a data race.  You may consider adding a method to the loader that would indicate that it finished.  It may even return a channel so that you can `select` on it.